build_tools/python/e2e_test_framework/unique_ids.py

```diff
@@ -3,11 +3,63 @@
 # Licensed under the Apache License v2.0 with LLVM Exceptions.
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-"""List of unique random IDs in the benchmark suites.
+"""List of unique random IDs in the framework and id utilities.
 
 Each ID should be generated from uuid.uuid4().
 """
 
+import hashlib
+from typing import Sequence
+
+# Special id which will be ignored when calculating the composite id.
+#
+# It should only be used when adding a new field to a composite object while
+# we want to maintain the same id on the existing composite objects.
+#
+# In such case, you need to create a "default config" for the new field with
+# this id and populate that config to the fields of the existing objects. The
+# composite id computing function will ignore this id and keep the output
+# unchanged.
+TRANSPARENT_ID = "75f74d68-9556-4c03-91ef-68c5b7f4bcbd"
```

Comment:
Why not use empty string or None for this sentinel value? That seems a bit clearer to me. I guess this way someone can't accidentally use it

Reply:
The "default config" still needs to have something in its id field, and can be serialized/deserialized. I suspect an empty ID might work? But might create a special case somewhere.

Reply:
Hmm, maybe using the special `00000000-0000-0000-0000-000000000000` is a good idea. It's easier to notice in the config file.

Reply:
Yeah I like that 🙂